shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/model/routeDetailsPage/RouteBranchingTest.kt

```diff
@@ -0,0 +1,354 @@
+                        ),
+                        BranchStop(backBay, StickState(left = empty, right = forward)),
+                        BranchStop(ruggles, StickState(left = empty, right = forward)),
+                    ),
+                    name = null,
+                    isTypical = true,
+                ),
+                BranchSegment(
+                    listOf(
+                        BranchStop(hydePark, StickState(left = empty, right = forward)),
+                        BranchStop(readville, StickState(left = empty, right = forward)),
+                    ),
+                    name = null,
+                    isTypical = false,
+                ),
+                BranchSegment(
+                    listOf(
+                        BranchStop(route128, StickState(left = empty, right = forward)),
+                        BranchStop(
+                            cantonJunction,
+                            StickState(left = crossTo.copy(before = false), right = crossFrom),
+                        ),
+                    ),
+                    name = null,
+                    isTypical = true,
+                ),
+                BranchSegment(
+                    listOf(
+                        BranchStop(cantonCenter, StickState(left = skip, right = forward)),
+                        BranchStop(
+                            stoughton,
+                            StickState(left = skip, right = forward.copy(after = false)),
+                        ),
+                    ),
+                    name = "Stoughton",
+                    isTypical = true,
+                ),
+                BranchSegment(
+                    listOf(
+                        BranchStop(sharon, StickState(left = forward, right = empty)),
+                        BranchStop(mansfield, StickState(left = forward, right = empty)),
+                        BranchStop(attleboro, StickState(left = forward, right = empty)),
+                    ),
+                    name = null,
+                    isTypical = true,
+                ),
+                BranchSegment(
+                    listOf(BranchStop(southAttleboro, StickState(left = forward, right = empty))),
+                    name = null,
+                    isTypical = false,
+                ),
+                BranchSegment(
+                    listOf(
+                        BranchStop(pawtucket, StickState(left = forward, right = empty)),
+                        BranchStop(providence, StickState(left = forward, right = empty)),
+                        BranchStop(tfGreenAirport, StickState(left = forward, right = empty)),
+                        BranchStop(
+                            wickfordJunction,
+                            StickState(left = forward.copy(after = false), right = empty),
+                        ),
+                    ),
+                    name = "Providence",
```

Comment:
suggestion: I think it would be worth some separate unit tests for the branch naming candidates to reflect each of the possible cases